src/creature.cpp

```diff
@@ -673,10 +674,32 @@ void Creature::onDeath()
 				mostDamage = cb.total;
 				mostDamageCreature = attacker;
 			}
-			attacker->onAttackedCreatureKilled(this);
+
+			if (attacker != this) {
+				uint64_t gainExp = getGainedExperience(attacker);
+				if (Player* player = attacker->getPlayer()) {
+					Party* party = player->getParty();
+					if (party && party->isSharedExperienceActive() && party->isSharedExperienceEnabled()) {
+						if (Player* leader = party->getLeader()) {
```

Comment:
Alright xD

```C++
if (party && party->getLeader() && party->isSharedExperienceActive() && party->isSharedExperienceEnabled()) {
    attacker = party->getLeader();
}
```
Should I change it to look like this, or is it supposed to be done in another way?